README.md

```diff
@@ -85,58 +114,89 @@ TimestampType -> long
+// then written back out as Avro records
+val df = sqlContext.read.avro("src/test/resources/episodes.avro")
+df.filter("doctor > 5").write.avro("/tmp/output")
 ```
 
-An alternative way is to use `avroFile` and `save` methods (available in 1.2+):
+Alternativly you can specify the format to use instead:
 
 ```scala
-scala> import org.apache.spark.sql.SQLContext
+val sqlContext = new SQLContext(sc)
+val df = sqlContext.read
+	.format("com.databricks.spark.avro")
+	.load("src/test/resources/episodes.avro")
+	
+df.filter("doctor > 5").write
+	.format("com.databricks.spark.avro")
+	.save("/tmp/output")
+```
 
-scala> val sqlContext = new SQLContext(sc)
+You can specify the compression like this:
 
-scala> import com.databricks.spark.avro._
+```scala
+import com.databricks.spark.avro._
+val sqlContext = new SQLContext(sc)
 
-scala> val episodes = sqlContext.avroFile("episodes.avro", 20)
-episodes: org.apache.spark.sql.DataFrame =
-DataFrame[0] at RDD at DataFrame.scala:104
-== Query Plan ==
-== Physical Plan ==
-PhysicalRDD [title#0,air_date#1,doctor#2], MappedRDD[2] at map at AvroRelation.scala:54
+// configuration to use deflate compression
+sqlContext.setConf("spark.sql.avro.compression.codec", "deflate")
+sqlContext.setConf("spark.sql.avro.deflate.level", "5")
 
-scala> import sqlContext._
-import sqlContext._
+val df = sqlContext.read.avro("src/test/resources/episodes.avro")
 
-scala> episodes.select('title).collect()
-res0: Array[org.apache.spark.sql.Row] = Array([The Eleventh Hour], [The Doctor's Wife], [Horror of Fang Rock], [An Unearthly Child], [The Mysterious Planet], [Rose], [The Power of the Daleks], [Castrolava])
+// writes out compressed Avro records
+df.write.avro("/tmp/output")
 ```
 
-`avroFile` allows you to specify the `minPartitions` parameter as its second argument.
-To save DataFrame as avro you should use the `save` method in `AvroSaver`. For example:
+You can write partitioned Avro records like this:
 
 ```scala
-scala> AvroSaver.save(myRDD, "my/output/dir")
+import com.databricks.spark.avro._
+
+val sqlContext = new SQLContext(sc)
+
+import sqlContext.implicits._
+
+val df = Seq((2012, 8, "Batman", 9.8),
+	(2012, 8, "Hero", 8.7),
+	(2012, 7, "Robot", 5.5),
+	(2011, 7, "Git", 2.0))
+	.toDF("year", "month", "title", "rating")
+
+df.write.partitionBy("year", "month").avro("/tmp/output")
 ```
-You can also specifiy the the record name and namespace with optional parameters:
+
+You can specify the record name and namespace like this:
+
 ```scala
-scala> AvroSaver.save(myRDD, "my/output/dir", Map("recordName" -> "MyRecord", "recordNamespace" -> "com.mycompany.mystuff"))
-```
+import com.databricks.spark.avro._
 
-We also support the ability to read all avro files from some directory. To do that, you can pass
-a path to that directory to the avroFile() function. However, there is a limitation - all of
-those files must have the same schema. Additionally, files used must have a .avro extension.
```

Comment:
Thanks rdblue, I will check out Kite. The usecase for us is that our avro schema evolves over time as we add or retire features. So some of our partitions will be stored with an older schema but if the newest schema is used to load them everything works fine with either Hive or base Spark. 

I might be insufficiently familiar with this library but since it appears as if there is no option to explicitly specify the load schema, there is no good way to load data across different partitions with compatible but different schemas.